### PR TITLE
Closes Issue #1601

### LINKED ISSUE #1601: Frequency of all elements in an array Java
Count the frequency of each element in the array
**Time Complexity O(n)
Auxiliary Space O(n)**

### PR DESCRIPTION
**Pull request by chanpreet1999**
Fixes #1601

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added New Code to count frequency of elements in array in Java

#### Languages Used:
-  Java

#### Files Added:
- CountFrequencyOfArrayElements.java

Thanks!

### CODE REVIEW

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/*Problem Statement
+ * Given an array of elements count the frequency of every element in the array.
+ */
+
+public class CountFrequencyOfArrayElements {
+
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		
+		//input size of array
```
Comment (MastersAbh) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Give 4 space indentation across the file

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import java.util.HashMap;
```
Comment (MastersAbh) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Please add problem statement on top of your file

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+				else
+					hm.put(a[i], 1);
+			}
+			
+		//display the frequencies	
+		System.out.println("Element \t Frequency");
+		hm.forEach(
+				(k,v) ->
+				{
+					System.out.println(k+" \t\t "+v);
```
Comment (MastersAbh) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Please fix indentation

**Comment by chanpreet1999** [comment]
Thanks for the reply can you show me an example of the required indentation, I would be highly obliged

**Comment by chanpreet1999** [comment]
I have tried to implement all the suggestions please do check and do let me know if there is room for improvement
Thank you for bearing with me

**Comment by chanpreet1999** [comment]
Hey actually I am new to all this. Can you help me do it.
As far as I know I should fork the repo and make a PR. But what aboug patch branching.
Plase guide me

**Comment by MastersAbh** [comment]
Hi @chanpreet1999 , It is all okay. The purpose of the event is to learn.
And 4 space indentation simply means, for every block that starts in the code, there should be 4 space extra margin than its parent block, like we write inside any loop, the contents of loop are basically 4 spaces ahead of the declaration of the loop

**Comment by jainaman224** [comment]
@chanpreet1999 Please use checkboxes from next time. I have edited the description.
Are you working on it?

**Comment by chanpreet1999** [comment]
yup, Actually I have an exam tomorrow , so I'll resolve this issues assigned to me tomorrow itself.Thanks for replying.

**Comment by chanpreet1999** [comment]
changes made @MastersAbh @jainaman224  
Thankyou for helping

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*Problem Statement
+ * Given an array of elements count the frequency of every element in the array.
+ */
+
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class CountFrequencyOfArrayElements {
+
+public static void main(String[] args) {
```
Comment (jainaman224) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Incorrect indentation.

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    for(int i = 0; i < n; i++)
+        a[i] = sc.nextInt();
+
+    countFreq(a, n);
+    sc.close();
+}
+
+//function to count frequency 
+private static void countFreq(int[] a, int n) {
+   // using hashmap to store the freq
```
Comment (jainaman224) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Inconsistent Indentation.

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    //input array
+    int a[] = new int[n];
+    for(int i = 0; i < n; i++)
+        a[i] = sc.nextInt();
+
+    countFreq(a, n);
+    sc.close();
+}
+
+//function to count frequency
```
Comment (jainaman224) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Add space after `//`.

**Comment by jainaman224** [comment]
@chanpreet1999 It would be great id you can ask on `proj_algodsnotes` channel on slack, if you are GSSoC aspirant.

**Comment by chanpreet1999** [comment]
> 
> 
> @chanpreet1999 It would be great id you can ask on `proj_algodsnotes` channel on slack, if you are GSSoC aspirant.

Yup I tried that too but didn't get a response, I have tried to figuree out on my own though. Hope this is okay now.

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            else
+                hm.put(a[i], 1);
+        }
+
+        //display the frequencies	
+        System.out.println("Element \t Frequency");
+        hm.forEach(
+        (k,v) ->
+        {
+            System.out.println(k+" \t\t "+v);
```
Comment (MastersAbh) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: Please add space around +

File: Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            if( hm.containsKey(a[i]) )
+                hm.put( a[i], hm.get(a[i]) + 1);
+            else
+                hm.put(a[i], 1);
+        }
+
+        //display the frequencies	
+        System.out.println("Element \t Frequency");
+        hm.forEach(
+        (k,v) ->
```
Comment (MastersAbh) [inline Freq_Of_Elements_Of_Array/CountFrequencyOfArrayElements.java]: give indentation for contents inside forEach. Shift them 4 spaces to right

**Comment by somya-kapoor** [comment]
@MastersAbh Please review the PR

**Pull request merged**